tests/unit/utils/disposable-test.js

```diff
@@ -2,26 +2,42 @@ import EmberObject from '@ember/object';
 import { run } from '@ember/runloop';
 import { module, test } from 'qunit';
 import {
-  registeredDisposables,
   registerDisposable,
   runDisposables,
-} from 'ember-lifeline/utils/disposable';
+  _setRegisteredDisposables,
+} from 'ember-lifeline';
 
 module('ember-lifeline/utils/disposable', function(hooks) {
   hooks.beforeEach(function() {
-    this.obj = EmberObject.create({
+    this.obj = EmberObject.extend({
       destroy() {
         runDisposables(this);
+
+        this._super(...arguments);
       },
-    });
+    }).create();
+
+    this.registeredDisposables = new Map();
+
+    _setRegisteredDisposables(this.registeredDisposables);
```

Comment:
Can you add an `hooks.after(() => _setRegisteredDisposables(new WeakMap())` (which will run after all tests in this module have completed) to ensure we properly reset back to a WeakMap?

Reply:
Done. Also reset in `afterEach`